Vagrant no longer autobuilds on provision

tobsan opened the pull request.

Commit:
Vagrant no longer autobuilds on provision

The autobuild step was mostly intended for a CI setup, so that part has
been moved to the Jenkinsfile instead.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
Jenkinsfile: better identation

Commit:
Jenkinsfile: try to expand params

kursatkobya reviewed (approved):
Jenkins build seems okay and changes looks nice

frznlogic commented on Jenkinsfile:
```diff
@@ -1,52 +1,55 @@
-    sh "cd ${workspace} && vagrant up --provision"
+    stage 'StartVM'
+        // Start the machine (destroy it if present) and provision it
+        sh "cd ${workspace} && vagrant destroy -f"
```
Will vagrant destroy -f always return 0 or should we or it with true?

tobsan replied:
Good point, will update.

frznlogic commented on Vagrantfile:
```diff
@@ -57,21 +57,4 @@ Vagrant.configure(2) do |config|
-            cd softwarecontainer
-            sudo su -c ./run-all-tests.sh
-        SHELL
-    end
```
Do we really want to delete all this, we dont want to keep it but with a if ENV thingy around it for "convenience"?

tobsan replied:
Sure, we could do the same with the building in the provision step.

Commit:
Retains the automatic building if env var set

If the env variable BUILD_AND_TEST is set, provisioning using the
Vagrantfile will still build, clang-analyze and test SoftwareContainer.

The old name CI_BUILD is not really descriptive any more, since in the
Jenkinsfile it is separated into several steps.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>
